### PR TITLE
split the README, resolving #315

### PR DESCRIPTION
**Pull request by shajoezhu**
split the README to separated .rst files for documentation

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Superb! This looks and works great @shajoezhu. We can elaborate on things later and update the various parts of the documentation later, but right now I think we should merge this much and test it to see if read-the-docs picks it up correctly.

+1.

**Comment by shajoezhu** [comment]
a temporary doc can be found at http://ga4gh-server-shajoezhu.readthedocs.org/en/latest/index.html

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,4 @@ target/
 
 # Pycharm
 .idea/
+docs/source/_build/
```
Comment (dcolligan) [inline .gitignore]: Please put this line at the top of the file with all the other project-specific excludes

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+
+# -- Options for Texinfo output -------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+  (master_doc, 'GA4GH', u'GA4GH Documentation',
+   author, 'GA4GH', 'One line description of project.',
```
Comment (dcolligan) [inline docs/source/conf.py]: should change 'One line description of project.' to something else

File: docs/source/Makefile [hunk 3]
```diff
@@ -0,0 +1,192 @@
+# Makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+PAPER         =
+BUILDDIR      = _build
```
Comment (dcolligan) [inline docs/source/Makefile]: is this the canonical sphinx build dir?  what gets stored here?  maybe we should choose a more descriptive name?

File: .gitignore (hunk 1 above)
Comment (dcolligan) [inline .gitignore]: Or, there is also a section for "Sphinx documentation" ... under that might also be desirable 

(but the existence of a exclusion for Sphinx already that is very similar to the exclusion we are adding makes me wonder if we shouldn't adhere to the existing convention...)

File: docs/source/Makefile [hunk 4]
```diff
@@ -0,0 +1,192 @@
+# Makefile for Sphinx documentation
```
Comment (dcolligan) [inline docs/source/Makefile]: can you add a comment about where you pulled this from?  I doubt you wrote all this yourself, right?

**Comment by jeromekelleher** [comment]
This is all standard sphinx boiler plate @dcolligan. When you run sphinx-quickstart it sets up all this stuff for you. We'll need to refine quite a lot of this as we go on, so I'm not really bothered if there's some details that aren't quite right.

**Comment by dcolligan** [comment]
Alright, since you guys seem to want to merge this quickly I'll put my objections aside for now.

**Comment by jeromekelleher** [comment]
Excellent, thanks @dcolligan. I want to make a push on the docs tomorrow and be ready to tag 0.1.0a3 when you get online. Hopefully it'll just be a case of checking things through on Friday, making a few final adjustments and tagging 0.1.0b1.

**Pull request merged**